frontend/packages/knative-plugin/src/components/add/event-sources/YAMLEditorSection.tsx

```diff
@@ -0,0 +1,23 @@
+import * as React from 'react';
+import { safeDump } from 'js-yaml';
+import { useFormikContext, FormikValues } from 'formik';
+import { YAMLEditorField, useFormikValidationFix } from '@console/shared';
+import FormSection from '@console/dev-console/src/components/import/section/FormSection';
+import { getEventSourcesDepResource } from '../../../utils/create-eventsources-utils';
+
+const YAMLEditorSection: React.FC = () => {
+  const { setFieldValue, setFieldTouched, values } = useFormikContext<FormikValues>();
+  const formData = React.useRef(values);
+  React.useEffect(() => {
+    setFieldValue('yamlData', safeDump(getEventSourcesDepResource(formData.current)));
+    setFieldTouched('yamlData', true);
+  }, [formData, setFieldTouched, setFieldValue]);
+  useFormikValidationFix(values.yamlData);
```

Comment:
@nemesis09 i think we discussed about moving it to handle change of `EventSourceSelector` right?

and set yamlData with values cc @rohitkrai03

Reply:
yes, but since the calls to setFieldValue are synchronised, the flow did not wait for the values to changes before moving to the next call, due to which we would have the yaml for the previously selected event source type
thus moved it back

Reply:
try Clicking on `pingSource` and see if it gets updated